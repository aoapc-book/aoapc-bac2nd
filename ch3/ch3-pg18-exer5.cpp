#include<stdio.h>

int main() {
    printf("Puzzle, ACM/ICPC World Finals 1993, UVa227\n");

#ifdef LOCAL
    freopen("data/ch3-pg18-exer5.in", "r", stdin);
#endif

    char puzzle[5][5] = {0};
    char *p = puzzle[0];
    char move = 0;
    int x = 0;
    int y = 0;
    for (int i = 0; i < 25; ++i) {
        scanf("%s", p++);
        if (*(p - 1) == '-') {
            x = i / 5;
            y = i % 5;
        }
    }
    while ((move = getchar()) != EOF && move != 0) {
        switch(move) {
            case 'A':
                puzzle[x][y] = puzzle[(x + 5 - 1) % 5][y];
                x = (x + 5 - 1) % 5;
                puzzle[x][y] = '-';
                break;
            case 'B':
                puzzle[x][y] = puzzle[(x + 1) % 5][y];
                x = (x + 1) % 5;
                puzzle[x][y] = '-';
                break;
            case 'L':
                puzzle[x][y] = puzzle[x][(y + 5 - 1) % 5];
                y = (y + 5 - 1) % 5;
                puzzle[x][y] = '-';
                break;
            case 'R':
                puzzle[x][y] = puzzle[x][(y + 1) % 5];
                y = (y + 1) % 5;
                puzzle[x][y] = '-';
                break;
            default:
                ;
        }
    }
    for (int i = 0; i < 5; ++i) {
        for (int j = 0; j < 5; ++j) {
            printf("%c ", puzzle[i][j]);
        }
        printf("\n");
    }
    return 0;
}
