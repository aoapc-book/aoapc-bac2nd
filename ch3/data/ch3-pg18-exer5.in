Q W E R T
A S D F G
Z - C V B
Y U I O P
H J K L M
ARRRRBBRBBRBL0
