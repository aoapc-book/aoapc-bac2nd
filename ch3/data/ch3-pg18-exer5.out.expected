Q E F R T
Z - D G A
Y S C V B
U J I O P
H W K L M
